homeassistant/components/sensor/sabnzbd.py

```diff
@@ -0,0 +1,142 @@
+"""
+homeassistant.components.sensor.sabnzbd
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Monitors SABnzbd NZB client API
+
+Configuration:
+
+To use the SABnzbd sensor you will need to add something like the following to
+your config/configuration.yaml
+
+sensor:
+    platform: sabnzbd
+    name: SAB
+    api_key: YOUR_API_KEY
+    base_url: YOUR_SABNZBD_BASE_URL
+    monitored_variables:
+        - type: 'current_status'
+        - type: 'speed'
+        - type: 'queue_size'
+        - type: 'queue_remaining'
+        - type: 'disk_size'
+        - type: 'disk_free'
+
+VARIABLES:
+
+base_url
+*Required
+This is the base URL of your SABnzbd instance including the port number if not
+running on 80
+Example: http://192.168.1.32:8124/
+
+
+name
+*Optional
+The name to use when displaying this SABnzbd instance
+
+monitored_variables
+*Required
+An array specifying the variables to monitor.
+
+These are the variables for the monitored_variables array:
+
+type
+*Required
+The variable you wish to monitor, see the configuration example above for a
+list of all available variables
+
+
+"""
+
+from homeassistant.helpers.device import Device
+# pylint: disable=no-name-in-module, import-error
+from homeassistant.external.nzbclients.sabnzbd import SabnzbdApi
+from homeassistant.const import (
+    ATTR_UNIT_OF_MEASUREMENT, ATTR_FRIENDLY_NAME)
+import logging
+
+
+SENSOR_TYPES = {
+    'current_status': ['Status', ''],
+    'speed': ['Speed', 'MB/s'],
+    'queue_size': ['Queue', 'MB'],
+    'queue_remaining': ['Left', 'MB'],
+    'disk_size': ['Disk', 'GB'],
+    'disk_free': ['Disk Free', 'GB'],
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the sensors """
+    api_key = config.get("api_key")
+    base_url = config.get("base_url")
+    name = config.get("name", "SABnzbd")
+    if not base_url:
+        _LOGGER.error('Missing config variable base_url')
+        return False
+    if not api_key:
+        _LOGGER.error('Missing config variable api_key')
+        return False
+
+    sab_api = SabnzbdApi(base_url, api_key)
```

Comment:
Would be great if we add some error checking to see if the base_url and api_key are valid. Otherwise you'll add 5 sensors and every time they update your log will be filled with failed connections.